DeploymentAgentReconfigurationScripts/TFS2018 Update2/HardwareMoveRemoveOldAgent.ps1

```diff
@@ -0,0 +1,178 @@
+<# 
+This is a guidance script to bring a deployment group agent online after TFS Hardware move activity for TFS2018 Update2. Post Hardware move, all the agents will be offline as the TFS server url has been updated. User can use\follow this script to bring them online. The script has to be run from Administrator PowerShell prompt on the each agent machine.
+
+Inputs:
+1. targetTFSUrl : New TFS url after hardware move.
+2. patToken : PAT token from new TFS with Deployment group manage scope (at least)
+3. agentDownloadUrl: this is optional, user can specify if she wants any specific agent version to be installed.
+4. existingAgentFolder: The script will Auto-detect the agent folder if it was running as windows service. User need to pass the folder path otherwise.
+5. action: By default, the script will not do any update operation, it will just print what changes will happen after the script is applied. User need to set action parameter to 'apply' to execute actual steps.
+
+Output:
+If action is set to apply, this script will
+1. Delete the existing unusable offline agent and re-configure a new agent in with same name and properties including tags.
+2. All existing deployment history will be deleted. 
+
+Example usage:
+.\HardwareMoveRemoveOldAgent.ps1 -targetTFSUrl <newTfsUrl> -patToken <PAT-for-new-TFS>
+.\HardwareMoveRemoveOldAgent.ps1 -targetTFSUrl <newTfsUrl> -patToken <PAT-for-new-TFS> -existingAgentFolder <AgentFoilder (C:\vstsagents\A1)>
+.\HardwareMoveRemoveOldAgent.ps1 -targetTFSUrl <newTfsUrl> -patToken <PAT-for-new-TFS> -existingAgentFolder <AgentFoilder (C:\vstsagents\A1)> -action 'apply'
+#>
+
+param([string]$targetTFSUrl,
+      [string]$patToken,
+      [string]$existingAgentFolder = "",
+      [string]$agentDownloadUrl = 'https://vstsagentpackage.azureedge.net/agent/2.131.0/vsts-agent-win-x64-2.131.0.zip',
+      [string]$action = "PrintEffect")
+
+$ErrorActionPreference="Stop"
+
+# Basic input validations
+If (-NOT ([Security.Principal.WindowsPrincipal][Security.Principal.WindowsIdentity]::GetCurrent() ).IsInRole( [Security.Principal.WindowsBuiltInRole] �Administrator�)){ 
+    throw "Run command in an administrator PowerShell prompt"
+};
+
+If ($PSVersionTable.PSVersion -lt (New-Object System.Version("3.0"))){
+    throw "The minimum version of Windows PowerShell that is required by the script (3.0) does not match the currently running version of Windows PowerShell."
+}
+
+
+# Auto detect existing agent folder
+if ($existingAgentFolder -eq ""){
+    $ServiceNamePrefix = 'vstsagent*'
+    $agentServices = Get-Service -Name $ServiceNamePrefix
+    if ($agentServices.Count -eq 1){
```

Comment:
if multiple agent service found, should we log and throw error accordingly

Reply:
I thought about it, but did it this way to keep the message simple user like. "What he need to do?" rather than why we are not able to detect the folder automatically. I am open to change it.